File: rust/proverb.rs
pub fn build_proverb(list: &[&str]) -> String {
    let mut the_prov: String = String::new();
    if !list.is_empty() {
        for i in 1..list.len() {
            the_prov += &format!("For want of a {} the {} was lost.\n", list[i - 1], list[i]);
        }
        the_prov += &format!("And all for the want of a {}.", list[0]);
    }
    the_prov
}

File: clojure/proverb.clj
(ns proverb)

(defn for-want [[wanted lost]]
  (str "For want of a " wanted " the " lost " was lost.\n"))

(defn recite [items]
  (if (empty? items) ""
      (str (clojure.string/join (map for-want (partition 2 1 items)))
           "And all for the want of a " (first items) ".")))

